src/librustc/util/ppaux.rs

```diff
@@ -1532,3 +1544,29 @@ impl<'tcx> Repr<'tcx> for ast::Unsafety {
         format!("{:?}", *self)
     }
 }
+
+// Wrapper around lookup_trait_def that can handle when the trait is
+// defined in the local crate. Returns Option<_> because this function
+// can be called while the trait_defs map is being built, so that
+// looking up the trait_def in the local crate can sometimes fail.
+fn try_lookup_trait_def<'tcx>(tcx: &ctxt<'tcx>, did: ast::DefId)
+                              -> Option<Rc<ty::TraitDef<'tcx>>> {
+    if did.krate == ast::LOCAL_CRATE {
+        tcx.trait_defs.borrow().get(&did).map(|r| r.clone())
+    } else {
+        Some(ty::lookup_trait_def(tcx, did))
+    }
+}
+
+// Wrapper around lookup_trait_def that can handle when the trait is
+// defined in the local crate. Returns Option<_> because this function
+// can be called while the trait_defs map is being built, so that
+// looking up the trait_def in the local crate can sometimes fail.
+fn try_lookup_trait_def<'tcx>(tcx: &ctxt<'tcx>, did: ast::DefId)
```

Comment:
Ugh I totally overlooked the fact that you seem to have defined the same function twice here.